node/lib/util/rebase.js

```diff
@@ -42,6 +42,18 @@ const SubmoduleUtil = require("./submodule_util");
 const UserError     = require("./user_error");
 
 /**
+ * Put the head of the specified `repo` on the specified `commitSha`.
+ */
+const setHead = co.wrap(function *(repo, commitSha) {
+    // TODO: use a more "gentle" strategy that won't stomp on untracked files.
+    // The checkout strategy I was using doesn't handle all situations well,
```

Comment:
This code is being called in two places; one of those places was using `checkout` previously.